sqlglot/optimizer/eliminate_subqueries.py

```diff
@@ -142,13 +142,21 @@ def _eliminate_derived_table(scope, existing_ctes, taken):
     if scope.parent.pivots or isinstance(scope.parent.expression, exp.Lateral):
         return None
 
-    parent = scope.expression.parent
+    # Get rid of redundant exp.Subquery expressions, i.e. those that are used as parentheses
+    to_replace = scope.expression.parent
+    while (
+        isinstance(to_replace, exp.Subquery)
+        and to_replace.same_parent
+        and not any(v is not None and k != "this" for k, v in to_replace.args.items())
+    ):
+        to_replace = to_replace.parent
```

Comment:
The idea here is that we don't want remove subqueries that have additional args, e.g. laterals, alias, joins, ...

Perhaps we can also aggregate these "special" args into a set, so that we can avoid checking _all_ query modifier args in `exp.Subquery` and just focus on them?

Reply:
can this just be implement in def unnest?

Reply:
Where do you mean?

Reply:
there's a method unnest that is supposed to do this behavior

Reply:
I'll take a look, perhaps we can do this "unnesting" at an earlier stage

Reply:
@tobymao I took a stab at this, can you take another look? Happy to pivot if this is not what you had in mind.